src/plugins/enzymes/src/ConstructMoleculeDialog.cpp

```diff
@@ -332,35 +332,30 @@ void ConstructMoleculeDialog::sl_onItemClicked(QTreeWidgetItem* item, int column
 
 void ConstructMoleculeDialog::sl_onAddFromProjectButtonClicked() {
     ProjectTreeControllerModeSettings settings;
+    settings.allowMultipleSelection = false;
     settings.objectTypesToShow.insert(GObjectTypes::SEQUENCE);
     QScopedPointer<U2SequenceObjectConstraints> seqConstraints(new U2SequenceObjectConstraints());
     seqConstraints->alphabetType = DNAAlphabet_NUCL;
     settings.objectConstraints.insert(seqConstraints.data());
-
     QList<GObject*> objects = ProjectTreeItemSelectorDialog::selectObjects(settings, this);
+    CHECK(!objects.isEmpty(), );
+    SAFE_POINT(objects.size() == 1, QString("Incorrect selected objects size, expected: 1, current: %1").arg(objects.size()), );
 
-    if (!objects.isEmpty()) {
-        foreach (GObject* obj, objects) {
-            if (obj->isUnloaded()) {
-                continue;
-            }
-            auto seqObj = qobject_cast<U2SequenceObject*>(obj);
-
-            if (seqObj) {
-                QObjectScopedPointer<CreateFragmentDialog> dlg = new CreateFragmentDialog(seqObj, U2Region(0, seqObj->getSequenceLength()), this);
-                dlg->exec();
-                CHECK(!dlg.isNull(), );
-
-                if (dlg->result() == QDialog::Accepted) {
-                    DNAFragment frag = dlg->getFragment();
-                    QString fragItem = QString("%1 (%2) %3").arg(frag.getSequenceName()).arg(frag.getSequenceDocName()).arg(frag.getName());
-                    fragments.append(frag);
-                    fragmentListWidget->addItem(fragItem);
-                    break;
-                }
-            }
-        }
-    }
+    auto obj = objects.first();
```

Comment:
The old code was very explicit to support multiple selection.
Why did you decide to change it to support only 1 element at a time?

Reply:
Current behaviout is weird and confusing - you can select several sequences, but the fhagment will be created just to the first one (because of the `break`, which ends the `for` cycle). So, obviously, we have two options:

1. Remove breake and make possible to create as many fragments as you want from the first try.
2. Make selection singular and user'll be avaliable to select only one sequnce - and create only one fragment.

Here you should understand, that the special "Create fragment" dialog should appear every selected sequence for. So, if you make multiple selection, several dialogs will appeare - one after another. If you select 10 sequence - 10 dialogs will appear. If select 20 - 20 dialogs. This is not good. Besides, this dialog has some strict view and not connected to any sequence - which means, that user can't even define, what sequence for he's creating a fragment in this dialog.

Reply:
Ok sounds reasonable